### DISCUSSION TITLE
How to implement compare-with-carry in sleigh

### DISCUSSION THREAD

**Discussion by EwTE-H** [NationalSecurityAgency/ghidra#6409] [Q&A]
I have no idea for the complicated logic...
Semantics:
CMPC reg1, reg2
reg1 - reg2 - carry
carry: goes to 1 if this operation produces a carry out of bit 7 and to 0 otherwise.
zero: remains 1 only if it was 1 before execution and the result is zero. Otherwise goes to 0.
sign: tracks the top bit of the result.
overflow: goes to one if the operation produces overflow and to 0 otherwise.

EDIT: I copied the code from processor 6502 but it didn't work correctly. The decompiler gave a messy output. For assembly as such
```
CMP R0, 0x32
CMPC R1, 0xfc
```
The decompiler said
```c
if (r0 == 0x32 && r1 == 0xfc + (r0 < 0x32))
```
While it is expected that
```c
if (er0 == 0xfc32)
```

**Comment by GhidorahRex** [comment]
What are the semantics for the instruction?

**Reply by EwTE-H** [reply]
edited
